ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnersTable.vue

```diff
@@ -567,11 +538,23 @@ export default defineComponent({
     // check if Owner Group that has deceased Owners is valid
     const isInvalidOwnerGroup = (groupId): boolean => {
       if ((isTransferToExecutorProbateWill.value || isTransferToExecutorUnder25Will.value) && props.validateTransfer) {
-        const hasRemovedOwners = TransWill.hasSomeOwnersRemoved(groupId)
-        const hasExecutors = TransWill.hasAddedExecutorsInGroup(groupId)
-        const hasRemovedAllOwners = TransWill.hasAllCurrentOwnersRemoved(groupId)
-        const hasValidDocs = TransWill.hasOwnersWithValidSupportDocs(groupId)
-        const hasOwnersWithoutDeathCert = !TransWill.isAllGroupOwnersWithDeathCerts(groupId)
+        const hasRemovedOwners = TransToExec.hasSomeOwnersRemoved(groupId)
+        const hasExecutors = TransToExec.hasAddedExecutorsInGroup(groupId)
+        const hasRemovedAllOwners = TransToExec.hasAllCurrentOwnersRemoved(groupId)
+        const hasValidDocs = TransToExec.hasOwnersWithValidSupportDocs(groupId)
+        const hasOwnersWithoutDeathCert = !TransToExec.isAllGroupOwnersWithDeathCerts(groupId)
+
+        const isInvalid = !(hasRemovedAllOwners && hasValidDocs && hasExecutors && hasOwnersWithoutDeathCert)
+
+        return hasRemovedOwners && isInvalid
+      }
+
+      if (isTransferToExecutorProbateWill.value && props.validateTransfer) {
+        const hasRemovedOwners = TransToExec.hasSomeOwnersRemoved(groupId)
+        const hasExecutors = TransToAdmin.hasAddedAdministratorsInGroup(groupId)
+        const hasRemovedAllOwners = TransToExec.hasAllCurrentOwnersRemoved(groupId)
+        const hasValidDocs = TransToExec.hasOwnersWithValidSupportDocs(groupId)
+        const hasOwnersWithoutDeathCert = !TransToExec.isAllGroupOwnersWithDeathCerts(groupId)
```

Comment:
Several of these appear identical in definition, could they be defined once vs defined in each condition perhaps?

Reply:
Yea, I agree, more refactoring is needed. It will be in one of upcoming PRs.